src/angle/index.ts

```diff
@@ -1,63 +1,90 @@
- * @param x An Angle
- */
-export const cos = liftUnary(Math.cos);
-
-/**
- * Returns the tangent of an angle.
- * @param x An Angle
- */
-export const tan = liftUnary(Math.tan);
-
-/**
- * Returns the arcsine of a number.
- * @param x A numeric expression.
- */
-export const asin = liftRet(Math.asin);
-
-/**
- * Returns the arc cosine (or inverse cosine) of a number.
- * @param x A numeric expression.
- */
-export const acos = liftRet(Math.acos);
-
-/**
- * Returns the arctangent of a number.
- * @param x A numeric expression.
- */
-export const atan = liftRet(Math.atan);
-
-/**
- * Returns the angle from the X axis to a point.
- *
- * @param x A numeric expression representing the cartesian x-coordinate.
- * @param y A numeric expression representing the cartesian y-coordinate.
- */
-export function atan2(x: number, y: number): Angle {
-  return radians(Math.atan2(x, y));
-}
+import {Arithmetic, NativeArithmetic} from '../arithmetic';
+import {Quantity, makeUnitFactory} from '../unit';
 
-function liftRet(f: (x: number) => number): (x: number) => Angle {
-  return x => radians(f(x));
-}
+export type Angle<NumberType = number> = Quantity<NumberType, dimension.Angle>;
+
+export function withValueType<NumberType>(arithmetic: Arithmetic<NumberType>) {
+  const {makeUnit} = makeUnitFactory(arithmetic);
+  const {toNative, sin, cos, tan, asin, acos, atan, atan2} = arithmetic;
+
+  class WithValueType {
+    private constructor() {}
+
+    static radians = makeUnit('rad', dimension.Angle);
+    static degrees = WithValueType.radians
+      .times(Math.PI)
+      .per(180)
+      .withSymbol('º');
+
+    static turns = WithValueType.radians
+      .times(2)
+      .times(Math.PI)
+      .withSymbol('τ');
+
+    /**
+     * Returns the sine of an angle.
+     * @param x An Angle
+     */
+    static sin = liftUnary(sin);
+
+    /**
+     * Returns the cosine of an angle.
+     * @param x An Angle
+     */
+    static cos = liftUnary(cos);
 
-function liftUnary(f: (x: number) => number): (x: Angle) => number {
-  return x => f(x.in(radians).amount);
+    /**
+     * Returns the tangent of an angle.
+     * @param x An Angle
+     */
+    static tan = liftUnary(tan);
+
+    /**
+     * Returns the arcsine of a number.
+     * @param x A numeric expression.
+     */
+    static asin = liftRet(asin);
+
+    /**
+     * Returns the arc cosine (or inverse cosine) of a number.
+     * @param x A numeric expression.
+     */
+    static acos = liftRet(acos);
+
+    /**
+     * Returns the arctangent of a number.
+     * @param x A numeric expression.
+     */
+    static atan = liftRet(atan);
+
+    /**
+     * Returns the angle from the X axis to a point.
+     *
+     * @param x A numeric expression representing the cartesian x-coordinate.
+     * @param y A numeric expression representing the cartesian y-coordinate.
+     */
+    static atan2(x: NumberType, y: NumberType): Angle<NumberType> {
+      // TODO: Ideally, we should not use toNative here
```

Comment:
For my understanding, this would require pushing down the geometric functions into Arithmetic as well, no? Maybe expand the comment slightly to make this explicit to other readers of the code.